sdk/identity/azure-identity/src/samples/Azure-Cache-For-Redis/Jedis/java/com/azure/jedis/sample/AuthenticateWithTokenCache.java

```diff
@@ -83,19 +84,18 @@ public static class TokenRefreshCache {
         private final TokenRequestContext tokenRequestContext;
         private final Timer timer;
         private volatile AccessToken accessToken;
-        private final Duration refreshOffset;
+        private final Duration maxRefreshOffset = Duration.ofMinutes(5);
+        private final Duration baseRefreshOffset = Duration.ofMinutes(2);
 
         /**
          * Creates an instance of TokenRefreshCache
          * @param tokenCredential the token credential to be used for authentication.
          * @param tokenRequestContext the token request context to be used for authentication.
-         * @param refreshOffset the refresh offset to use to proactively fetch a new access token before expiry time.
          */
-        public TokenRefreshCache(TokenCredential tokenCredential, TokenRequestContext tokenRequestContext, Duration refreshOffset) {
+        public TokenRefreshCache(TokenCredential tokenCredential, TokenRequestContext tokenRequestContext) {
```

Comment:
@samsaha-ms 
Thank you for the feedback.
I've updated the token cache sample for Jedis scenario, to proactively authenticate the connection before token expires.

For other libraries (Lettuce and Redisson), proactively authenticating the connection isn't possible, as the authentication API/logic is not exposed on the client and kept internal. However, these libraries do internally retry and will fetch a new token from the token cache to re-authenticate. We do ensure that a fresh token is always available in the token cache samples for the library to use.

Reply:
For Lettuce can't we use this https://github.com/lettuce-io/lettuce-core/blob/afeb1e70eed7c3c49187e48318ea34ec9db81e1e/src/main/java/io/lettuce/core/api/sync/RedisCommands.java#LL51C1-L51C57?

For Redisson, I agree it would be difficult with RedissonClient interface. There is another way to connect and via using Redisson.getConfig().useSingleServer().setPassword("newToken") is possible to refresh token proactively https://github.com/redisson/redisson/blob/af9208c3c9951b7743585b25a7f6a3fcd69c6fe8/redisson/src/main/java/org/redisson/Redisson.java#L715. But this is not a recommended way to use this library so we can skip it.